test/test-plugin-loader.ts

```diff
@@ -13,430 +13,329 @@
-  });
 
-  /**
-   * Loads a module with internal exports and patches them, and then makes sure
-   * that they are actually patched.
-   */
-  it('intercepts internal files in modules', function() {
-    addModuleMock('module-h', '1.0.0', {
-      createSentence: function() {
-        return require('module-h/subject').get() + ' ' +
-          require('module-h/predicate').get() + '.';
-      }
+    it('doesn\'t patch modules for which plugins aren\'t specified', () => {
+      makePluginLoader({plugins: {}}).activate();
+      assert.strictEqual(require('small-number').value, 0);
     });
-    addModuleMock('module-h/subject', '', {
-      get: function() {
-        return 'bad tests';
-      }
+
+    it('patches modules when activated, with no plugin file field specifying the main file',
+       () => {
+         makePluginLoader({
+           plugins: {'small-number': 'plugin-small-number'}
+         }).activate();
+         assert.strictEqual(require('small-number').value, 1);
+         // Make sure requiring doesn't patch twice
+         assert.strictEqual(require('small-number').value, 1);
+         assert.strictEqual(
+             logger.getNumLogsWith('info', '[small-number@0.0.1]'), 1);
+       });
+
+    it('accepts absolute paths in configuration', () => {
+      makePluginLoader({
+        plugins: {'small-number': `${SEARCH_PATH}/plugin-small-number`}
+      }).activate();
+      assert.strictEqual(require('small-number').value, 1);
+      assert.strictEqual(
+          logger.getNumLogsWith('info', '[small-number@0.0.1]'), 1);
     });
-    addModuleMock('module-h/predicate', '', {
-      get: function() {
-        return 'don\'t make sense';
-      }
+
+    it('unpatches modules when deactivated', () => {
+      const loader = makePluginLoader({
+                       plugins: {'small-number': 'plugin-small-number'}
+                     }).activate();
+      require('small-number');
+      loader.deactivate();
+      assert.strictEqual(require('small-number').value, 0);
+      // One each for activate/deactivate
+      assert.strictEqual(
+          logger.getNumLogsWith('info', '[small-number@0.0.1]'), 2);
     });
-    addModuleMock('module-h-plugin', '', [
-      {
-        file: 'subject',
-        intercept: function(originalModule, api) {
-          return {
-            get: function() {
-              return 'good tests';
-            }
-          };
-        }
-      },
-      {
-        file: 'predicate',
-        intercept: function(originalModule, api) {
-          return {
-            get: function() {
-              return 'make sense';
-            }
-          };
-        }
-      }
-    ]);
-    assert.strictEqual(require('module-h').createSentence(),
-      'bad tests don\'t make sense.');
-    // Activate plugin loader
-    pluginLoader.activate(logger, {
-      plugins: {
-        'module-h': 'module-h-plugin'
-      }
+
+    it('doesn\'t unpatch twice', () => {
```

Comment:
Nit. Better to use a template string instead of escaping quotes.

Reply:
Done